src/rust/engine/process_execution/src/docker.rs

```diff
@@ -185,13 +196,34 @@ async fn pull_image(docker: &Docker, image: &str, policy: ImagePullPolicy) -> Re
     let mut result_stream = docker.create_image(Some(create_image_options), None, None);
     while let Some(msg) = result_stream.next().await {
       log::trace!("pull {}: {:?}", image, msg);
-      if let Err(err) = msg {
-        return Err(format!(
-          "Failed to pull Docker image `{}`: {:?}",
-          image, err
-        ));
-      }
+      match msg {
+        Ok(msg) => match msg {
+          CreateImageInfo {
+            error: Some(error), ..
+          } => {
+            let error_msg = format!("Failed to pull Docker image `{image}`: {error}");
+            log::error!("{error_msg}");
+            return Err(error_msg);
+          }
+          CreateImageInfo {
+            status: Some(status),
+            ..
+          } => {
+            log::info!("Docker pull status: {status}");
```

Comment:
Maybe this is better as `debug`?

Reply:
These message are what the user would see when running `docker pull` manually. I'd like to provide the same UX for the moment. If its too chatty, we can always put it to debug level.